vunit/ui/results.py

```diff
@@ -24,5 +29,62 @@ def merge_coverage(self, file_name, args=None):
         :param file_name: The resulting coverage file name.
         :param args: The tool arguments for the merge command. Should be a list of strings.
         """
-
         self._simulator_if.merge_coverage(file_name=file_name, args=args)
+
+    def get_report(self):
+        """
+        Get a report (dictionary) of tests: status, time and output path
+        """
+
+        def get_relative_test_path(test_path):
+            """
+            If the test_path is relative to the default TEST_OUTPUT_PATH, return the subdir only
+            """
+            base = basename(test_path)
+            return (
+                base
+                if normpath(join(self._output_path, TEST_OUTPUT_PATH, base))
+                == normpath(test_path)
+                else test_path
+            )
+
+        report = Report(self._output_path)
+        for (
+            test
+        ) in self._report._test_results_in_order():  # pylint: disable=protected-access
+            obj = test.to_dict()
+            report.tests.update(
+                {
+                    test.name: ReportResult(
+                        obj["status"], obj["time"], get_relative_test_path(obj["path"])
+                    )
+                }
+            )
+        return report
+
+
+class Report(object):
+    """
+    Gives access to test results and paths after running tests
+    """
+
+    def __init__(self, output_path):
+        self.output_path = output_path
+        self.tests = {}
+
+    def get_full_path(self, test_path, file_path=""):
+        """
+        Return the full path to a test subdir or to a file within it
+        """
+        return join(self.output_path, TEST_OUTPUT_PATH, test_path, file_path)
+
+
+class ReportResult(object):
+    """
```

Comment:
TestResult is a better name

Reply:
`TestResult` is already used in `vunit/test_report.py`. I tried not to repeat the name of the class, even though there is no collision because this is located in submodule `vunit.ui`. Should I change it anyway?

Reply:
Yes I think it is a better name. As you say it is not a name conflict.